src/lib/MapCards.svelte

```diff
@@ -29,9 +31,14 @@
   const setPreviewKey = (key: string | null) => (previewKey = key)
 
   onMount(async () => {
+    toggleVisibility.set(true)
     await getMaps()
   })
 
+  onDestroy(() => {
+    toggleVisibility.set(false)
```

Comment:
This could cause issues in theory. Issue scenario would be:

- You have two map cards (like verified and curated on the main page)
- You dismount one of them

You could handle that case by making it a count instead (increment on mount, decrement on dismount/destroy). In turn you could check whether it is more than zero or you could derive a new store from it which will make the same check, but hide the logic.

This is however not an issue I forsee happening for us.